### PR TITLE
refactores $GLOBALS['TYPO3_DB'] to QueryBuilder

### PR DESCRIPTION
**Pull request by erik-konrad**
This pull request refactores $GLOBALS['TYPO3_DB'] to QueryBuilder functionality for TYPO3 9LTS compatibility.
* all methods are refactored for QueryBuilder usage
* Helper::whereClause() removed (check for hidden and deleted is default restriction configuration in querybuilder)
* general code cleanup

### CODE REVIEW

File: class.ext_update.php [hunk 1]
```diff
@@ -116,26 +91,86 @@ public function main()
...
+            ->select('tx_dlf_formats.uid AS uid', 'tx_dlf_formats.type AS type')
+            ->from('tx_dlf_formats')
+            ->where('tx_dlf_formats.class IS NOT NULL AND tx_dlf_formats.class != "" AND tx_dlf_formats.class NOT LIKE "%\\\\\\\\%"')
+            ->execute();
+
+        while ($resArray = $result->fetch()) {
             $oldRecords[$resArray['uid']] = $resArray['type'];
         }
+
         return $oldRecords;
```
Comment (albig) [inline class.ext_update.php:177]: Actually this Query doesn't work as expected. As "oldFormats", also current ones are found:
```
array (size=3)
  1 => string 'MODS' (length=4)
  2 => string 'TEIHDR' (length=6)
  3 => string 'ALTO' (length=4)
```
This is set by Kitodo.Presentation 3.0 and is working on some installations:
```
MariaDB [typo3]> select uid,type,class from tx_dlf_formats WHERE class IS NOT NULL AND class != "";
+-----+--------+--------------------------------+
| uid | type   | class                          |
+-----+--------+--------------------------------+
|   1 | MODS   | Kitodo\\Dlf\\Format\\Mods      |
|   2 | TEIHDR | Kitodo\\Dlf\\Format\\TeiHeader |
|   3 | ALTO   | Kitodo\\Dlf\\Format\\Alto      |
+-----+--------+--------------------------------+
3 rows in set (0.001 sec)
```

File: class.ext_update.php [hunk 2]
```diff
@@ -38,61 +40,34 @@ class ext_update
...
+        }
+        if ($this->oldIndexRelatedTableNames()) {
             return true;
-        } elseif ($this->solariumSolrUpdateRequired()) {
+        }
+        if ($this->solariumSolrUpdateRequired()) {
             return true;
-        } elseif (count($this->oldFormatClasses())) {
+        }
+        if (count($this->oldFormatClasses())) {
```
Comment (albig) [inline class.ext_update.php:54]: This test will always be run --> see below in https://github.com/kitodo/kitodo-presentation/pull/499/files#r404744493

File: class.ext_update.php [hunk 3]
```diff
@@ -375,58 +452,27 @@ protected function doSolariumSolrUpdate()
...
-        $result = $GLOBALS['TYPO3_DB']->sql_query($sqlQuery);
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tx_dlf_documents');
+
+        $result = $queryBuilder
+            ->update('tx_dlf_documents')
+            ->where(
+                $queryBuilder->expr()->orX(
+                    $queryBuilder->expr()->eq('document_format', $queryBuilder->createNamedParameter(null)),
+                    $queryBuilder->expr()->eq('document_format', $queryBuilder->createNamedParameter(''))
+                ),
```
Comment (albig) [inline class.ext_update.php]: This comma (`,`) breaks TYPO3 8.7 ;-)
```
syntax error, unexpected ')'
```

File: ext_emconf.php [hunk 4]
```diff
@@ -13,12 +13,12 @@
 $EM_CONF[$_EXTKEY] = [
     'title' => 'Kitodo.Presentation',
     'description' => 'Base plugins, modules, services and API of the Digital Library Framework. It is part of the community-based Kitodo Digitization Suite.',
-    'version' => '3.0.0',
+    'version' => '3.0.1',
```
Comment (albig) [inline ext_emconf.php]: Shouldn't we rise to 3.1.0 already? To avoid problem with next release?

**Review by albig — CHANGES_REQUESTED** [review]
The one query may be wrong in current code, but should be fixed.

Without the mentioned comma, the code is running in 9.5 and 8.7. But I haven't tested the update queries itself.

File: class.ext_update.php (hunk 1 above)
Comment (erik-konrad) [reply class.ext_update.php:177]: fixed

File: class.ext_update.php (hunk 2 above)
Comment (erik-konrad) [reply class.ext_update.php:54]: also fixed

File: class.ext_update.php (hunk 3 above)
Comment (erik-konrad) [reply class.ext_update.php]: yes :-) thats the reason why TYPO3 is named TYPO3 :-D a simple syntax error. fixed.

File: ext_emconf.php (hunk 4 above)
Comment (erik-konrad) [reply ext_emconf.php]: done.

File: class.ext_update.php [hunk 5]
```diff
@@ -116,26 +91,89 @@ public function main()
...
-        while ($resArray = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($result)) {
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tx_dlf_formats');
+
+        $result = $queryBuilder
+            ->select('tx_dlf_formats.uid AS uid', 'tx_dlf_formats.type AS type')
+            ->from('tx_dlf_formats')
+            ->where(
+                $queryBuilder->expr()->isNotNull('tx_dlf_formats.class'),
+                $queryBuilder->expr()->neq('tx_dlf_formats.class', $queryBuilder->createNamedParameter('')),
+                $queryBuilder->expr()->notLike('tx_dlf_formats.class', $queryBuilder->createNamedParameter('%\\\\%'))
```
Comment (albig) [inline class.ext_update.php]: This looks better but still does not work. The below values are correct and shouldn't be touched.
```
MariaDB [typo3]> select uid,type,class from tx_dlf_formats WHERE class IS NOT NULL AND class != "" AND class NOT LIKE '%\\\\%';
+-----+--------+--------------------------------+
| uid | type   | class                          |
+-----+--------+--------------------------------+
|   1 | MODS   | Kitodo\\Dlf\\Format\\Mods      |
|   2 | TEIHDR | Kitodo\\Dlf\\Format\\TeiHeader |
|   3 | ALTO   | Kitodo\\Dlf\\Format\\Alto      |
+-----+--------+--------------------------------+
3 rows in set (0.001 sec)
```
The check should only return old formats like "tx_dlf_alto" (or how this old ALTO class was named).
Maybe @sebastian-meyer could explain the history.

**Review by albig — CHANGES_REQUESTED** [review]

File: class.ext_update.php (hunk 5 above)
Comment (erik-konrad) [reply class.ext_update.php]: Thats weird. If I run this query, I get the following result:

`
MariaDB [kitodo9]> select uid,type,class from tx_dlf_formats WHERE class IS NOT NULL AND class != "" AND class NOT LIKE '%\\\\%';
Empty set (0.000 sec)
`

Comment (erik-konrad) [reply class.ext_update.php]: I changed the part `NOT LIKE '%\\\\%'` to `LIKE '%tx_dlf_%'`. Now this should work as expected.

**Review by albig — APPROVED** [review]
Works for me on 8.7 and 9.5 now.

File: class.ext_update.php [hunk 6]
```diff
@@ -116,26 +91,89 @@ public function main()
...
-        while ($resArray = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($result)) {
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tx_dlf_formats');
+
+        $result = $queryBuilder
+            ->select('tx_dlf_formats.uid AS uid', 'tx_dlf_formats.type AS type')
+            ->from('tx_dlf_formats')
+            ->where(
+                $queryBuilder->expr()->isNotNull('tx_dlf_formats.class'),
+                $queryBuilder->expr()->neq('tx_dlf_formats.class', $queryBuilder->createNamedParameter('')),
+                $queryBuilder->expr()->like('tx_dlf_formats.class', $queryBuilder->createNamedParameter('%tx_dlf_%'))
```
Comment (sebastian-meyer) [inline class.ext_update.php:170]: This won't work for users which use custom classes for their format handling. But I don't know of anybody who does this currently, so we should be fine anyways.

**Review by sebastian-meyer — APPROVED** [review]

**Pull request merged**